Production/plugins/SummaryProducer.cc

```diff
@@ -88,8 +88,6 @@ class SummaryProducer : public edm::EDAnalyzer {
         (*expressTuple)().lumi = event.id().luminosityBlock();
         (*expressTuple)().evt = event.id().event();
         (*expressTuple)().genEventWeight = genEvent->weight();
-        (*expressTuple)().gen_top_pt = ntuple::DefaultFillValue<Float_t>();
-        (*expressTuple)().gen_topBar_pt = ntuple::DefaultFillValue<Float_t>();
```

Comment:
default values should still be set in case if saveGenTopInfo is false. keep these lines